doc/modules/grid_search.rst

```diff
@@ -231,6 +231,201 @@ some parameter settings could be fully evaluated. Setting ``error_score=0``
+iterations of a random forest::
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.model_selection import GridSuccessiveHalving
+    >>> import pandas as pd
+    >>>
+    >>> parameters = {'max_depth': [3, 5, 10],
+    ...               'min_samples_split': [2, 5, 10]}
+    >>> base_estimator = RandomForestClassifier(random_state=0)
+    >>> X, y = make_classification(n_samples=1000, random_state=0)
+    >>> sh = GridSuccessiveHalving(base_estimator, parameters, cv=5,
+    ...                            ratio=2,
+    ...                            budget_on='n_estimators',
+    ...                            max_budget=30,
+    ...                            random_state=0,
+    ...                            ).fit(X, y)
+    >>> sh.best_estimator_
+    RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
+                           max_depth=5, max_features='auto', max_leaf_nodes=None,
+                           min_impurity_decrease=0.0, min_impurity_split=None,
+                           min_samples_leaf=1, min_samples_split=2,
+                           min_weight_fraction_leaf=0.0, n_estimators=8,
+                           n_jobs=None, oob_score=False, random_state=0, verbose=0,
+                           warm_start=False)
+
+Note that it is not possible to budget on a parameter that is part of the
+parameter space.
+
+Exhausting the budget
+---------------------
+
+As mentioned above, the first iteration uses ``r_min`` resources. If you have
+a big budget, this may be a waste of resource::
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.svm import SVC
+    >>> from sklearn.model_selection import GridSuccessiveHalving
+    >>> import pandas as pd
+    >>> parameters = {'kernel': ('linear', 'rbf'),
+    ...               'C': [1, 10, 100]}
+    >>> base_estimator = SVC(gamma='scale')
+    >>> X, y = make_classification(n_samples=1000)
+    >>> sh = GridSuccessiveHalving(base_estimator, parameters, cv=5,
+    ...                            ratio=2).fit(X, y)
+    >>> results = pd.DataFrame.from_dict(sh.cv_results_)
+    >>> results.groupby('iter').r_i.unique()
+    iter
+    0    [20]
+    1    [40]
+    2    [80]
+    Name: r_i, dtype: object
+
+The search process will only use 80 resources at most, while our maximum budget
+is ``n_samples=1000``. Note in this case that ``r_min = r_0 = 20``. In order
+for the last iteration to use as many resources as possible, you can use the
+``force_exhaust_budget`` parameter::
+
+    >>> sh = GridSuccessiveHalving(base_estimator, parameters, cv=5,
+    ...                            ratio=2, force_exhaust_budget=True,
+    ...                            ).fit(X, y)
+    >>> results = pd.DataFrame.from_dict(sh.cv_results_)
+    >>> results.groupby('iter').r_i.unique()
+    iter
+    0     [250]
```

Comment:
Why is this a list? Are there iterations with more than one `r_i`?

Reply:
That's a pandas thing.

>Are there iterations with more than one r_i

No, each iteration has exactly one r_i.